bull-common/src/test/java/com/hotels/transformer/utils/ReflectionUtilsTest.java

```diff
@@ -153,6 +154,23 @@ public void testGetFieldValueWithAGivenFieldWorksAsExpected() throws Exception {
         };
     }
 
+    /**
+     * Tests that the method {@link ReflectionUtils#getFieldValue(Object, String, Class) getFieldValue} catches a runtime exception as InvalidBeanException.
+     */
+    @Test
+    public void testGetFieldValueCatchesRuntimeExceptionAsInvalidBeanException() {
+        // GIVEN
+        MutableToFoo mutableToFoo = createMutableToFoo(null);
+        ReflectionUtils underTestMock = Mockito.spy(ReflectionUtils.class);
```

Comment:
I think this is not needed, you can use the already existing `underTest ` variable

Reply:
We can't use the existing `underTest` variable since it is not a mocked object. If we do the following : 

`when(underTest.getDeclaredFieldType(LIST_FIELD_NAME, MutableToFoo.class)).thenThrow(new RuntimeException());`

it throws the following exception : 

```
org.mockito.exceptions.misusing.MissingMethodInvocationException: 
when() requires an argument which has to be 'a method call on a mock'.
For example:
    when(mock.getArticles()).thenReturn(articles);
```

Hence I had to create a spy so that I can mock the methods I want to and let it call its actual methods in other cases.